### ISSUE TITLE
cholesky_cuda: U(31,31) is zero, singular U.

### ISSUE THREAD

**Issue by Daisy-GENG** [ldeecke/gmm-torch#25]
Hi,

When using "full" covariance to fit the gmm model, the following error occurred:

```
cholesky_cuda: U(31,31) is zero, singular U.
  File "/Cylinder3D/network/gmm.py", line 320, in _calculate_log_det
    log_det[k] = 2 * torch.log(torch.diagonal(torch.linalg.cholesky(var[0,k]))).sum()
  File "/Cylinder3D/network/gmm.py", line 290, in _estimate_log_prob
    log_det = self._calculate_log_det(precision)
  File "/Cylinder3D/network/gmm.py", line 405, in __score
    weighted_log_prob = self._estimate_log_prob(x) + torch.log(self.pi)
  File "/Cylinder3D/network/gmm.py", line 170, in fit
    self.log_likelihood = self.__score(x)
  File "/Cylinder3D/train_gmm.py", line 72, in main
    gmm_model.fit(logit_cls)
  File "/Cylinder3D/train_gmm.py", line 88, in <module>
    main()
```

Is is related to my data or the computation accuracy? Is there any solution to this problem? 

Thank you so much!!

Best regards,
Daisy
